neuronunit/models/backends.py

```diff
@@ -423,28 +402,127 @@ def inject_square_current(self, current):
         self.h(prefix+'amplitude=%s'%amps)
         self.h(prefix+'duration=%s'%c['duration'])
         self.h(prefix+'delay=%s'%c['delay'])
-        self.local_run()
 
+        ##
+        # Duplicated call to run?
+        self.local_run()
+        ##
     def local_run(self):
         self.h('run()')
         results={}
         results['vm'] = [float(x/1000.0) for x in copy.copy(self.neuron.h.v_v_of0.to_python())]
-        results['t'] = [float(x) for x in copy.copy(self.neuron.h.v_time.to_python())]
+        results['t'] = [float(x/1000.0) for x in copy.copy(self.neuron.h.v_time.to_python())]
         if 'run_number' in results.keys():
             results['run_number'] = results['run_number']+1
         else:
             results['run_number'] = 1
         return results
 
-    def get_spike_train(self):
+
+
+    def get_spike_count(self):
         import neuronunit.capabilities.spike_functions as sf
-        return sf.get_spike_train(self.get_membrane_potential())
+        return len(sf.get_spike_train(self.get_membrane_potential()))
 
     def get_APs(self):
         import neuronunit.capabilities.spike_functions as sf
         return sf.get_spike_waveforms(self.get_membrane_potential())
 
```

Comment:
@russelljjarvis 
Neither of the two methods above should be there, as they should be handled at the level of the model, not the backend (at least not until `local_run`).  

For example, when the model calls `get_spike_count`, it will (according to the code in the `ProducesSpikes` capability) call `get_spike_train`.  `get_spike_train` is already implemented in `ReducedModel`.  It called `get_membrane_potential` which is also already implemented in `ReducedModel`.  It calls `run` which is implemented in `LEMSModel`, which called `local_run` which is implemented in the `NEURONBackend`.  Similary for `get_APs`.  

So see what happens when you get rid of these two methods here (`get_spike_count` and `get_APs`).

Reply:
Those methods are not available. I am unsure how often I should check back to see when this problem with method resolution order is fixed.

I propose that this issue be addressed with a unit test with the following idiom.
```
from .base import *
class TestsTestCase(object):
    """Abstract base class for testing tests"""
    def setUp(self):
        from neuronunit.models.reduced import ReducedModel
        from .model_tests import ReducedModelTestCase
        path = ReducedModelTestCase().path
        self.model = ReducedModel(path, backend='NEURON')

class TestBackend(TestTestCase, unittest.TestCase):
    def test_backend(self):
        method_avail = list(dir(self.model))        
        self.assertTrue('get_spike_train' in model_avail)
```

It's not productive for me to remove those methods from the backend until a unit test based on this idea can pass.

Reply:
@russelljjarvis 
OK, why don't you write a test like that (and set it to skip so it doesn't fail for you), and then after we merge I can turn off skipping and do what is required (if anything) to make it pass, and then we can remove those extra methods.

Reply:
AOK I have implemented this unit test. 

https://github.com/russelljjarvis/neuronunit/blob/dev/unit_test/testNEURONparallel.py#L44-L54

Reply:
The unittest passes, so I removed the aforementioned methods, `get_spike_count` and `get_APs`, which are appropriately inherited as required.

Reply:
Awesome!